src/compiler/moduleSpecifiers.ts

```diff
@@ -700,14 +707,45 @@ function getAllModulePaths(
         const cached = cache.get(importingFilePath, importedFilePath, preferences, options);
         if (cached?.modulePaths) return cached.modulePaths;
     }
-    const modulePaths = getAllModulePathsWorker(info, importedFileName, host);
+    const modulePaths = getAllModulePathsWorker(info, importedFileName, host, compilerOptions, options);
     if (cache) {
         cache.setModulePaths(importingFilePath, importedFilePath, preferences, options, modulePaths);
     }
     return modulePaths;
 }
 
-function getAllModulePathsWorker(info: Info, importedFileName: string, host: ModuleSpecifierResolutionHost): readonly ModulePath[] {
+const runtimeDependencyFields = ["dependencies", "peerDependencies", "optionalDependencies"] as const;
+
+function getAllRuntimeDependencies(packageJson: PackageJsonPathFields) {
+    let result;
+    for (const field of runtimeDependencyFields) {
+        const deps = packageJson[field];
+        if (deps && typeof deps === "object") {
+            result = concatenate(result, getOwnKeys(deps));

```

Comment:
You can probably just use `append` here, given `getOwnKeys` is going to give a fresh copy anyway.

Reply:
Eh, wrong argument type. Nevermind.

Reply:
You'd want `addRange`, not `append`, since `append` doesn't handle an array for input or take a spread list of arguments. But even that's not perfect - `concatenate` unconditionally allocates a new output array if both inputs are present (which is actually going to be rare here, usually it'll just be `dependencies`) - meanwhile `addRange` unconditionally copies `from` if `to` is *missing* (so it guarantees returning either `to` to a new array). That means it'd copy the first array needlessly every time. We actually don't have a "use either array mutably if they're present, it's fine"-type array-merging helper.

Honestly, it probably doesn't matter, though. Resizing one temporary array to merge in another one is probably going to trigger the similar allocations in the VM that making a new array would anyway (if they're both reasonably large). It's all short-lived objects.